src/components/Inventory/view/WithdrawalModal.tsx

```diff
@@ -259,82 +259,52 @@ export default function WithdrawalsModal(props: WithdrawalsModalProps): JSX.Elem
-                          id='withdrawnQuantity'
-                          value={withdrawQuantity}
-                          onChange={onChange}
-                          type='text'
-                          label={strings.WITHDRAW_QUANTITY}
-                          display={true}
-                        />
-                      </Grid>
-                    </>
-                  );
-                })}
+                {record.batchWithdrawals.map((bw, index) => (
+                  <Grid container key={`batch-not-ready-${index}`} sx={{ marginLeft: theme.spacing(2) }}>
+                    <Grid item xs={gridSize()} sx={marginTop} paddingRight={paddingSeparator}>
+                      <Textfield
+                        id='notReadyQuantityWithdrawn'
+                        value={bw.notReadyQuantityWithdrawn}
+                        onChange={(id, value) => onChangeQuantity('notReadyQuantityWithdrawn', value)}
+                        type='text'
+                        label={strings.NOT_READY_QUANTITY_REQUIRED}
+                        errorText={validateFields && bw.notReadyQuantityWithdrawn === 0 ? strings.REQUIRED_FIELD : ''}
+                      />
+                    </Grid>
+                    <Grid item xs={gridSize()} sx={marginTop} paddingLeft={paddingSeparator}>
+                      <Textfield
+                        id='readyQuantityWithdrawn'
+                        value={bw.readyQuantityWithdrawn}
+                        onChange={(id, value) => onChangeQuantity('readyQuantityWithdrawn', value)}
+                        type='text'
+                        label={strings.READY_QUANTITY_REQUIRED}
+                        errorText={validateFields && bw.readyQuantityWithdrawn === 0 ? strings.REQUIRED_FIELD : ''}
+                      />
+                    </Grid>
+                    <Grid item xs={gridSize()} sx={marginTop} paddingRight={paddingSeparator}>
+                      <Textfield
+                        id='withdrawnQuantity'
+                        value={withdrawQuantity}
+                        onChange={onChange}
+                        type='text'
+                        label={strings.WITHDRAW_QUANTITY}
+                        display={true}
+                      />
+                    </Grid>
+                    <Grid item xs={gridSize()} sx={marginTop} paddingLeft={paddingSeparator}>
+                      <DatePicker
+                        id='withdrawnDate'
+                        label={strings.WITHDRAW_DATE_REQUIRED}
+                        aria-label={strings.WITHDRAW_DATE_REQUIRED}
+                        value={record.withdrawnDate}
+                        onChange={changeDate}
+                        errorText={validateFields && !record.withdrawnDate ? strings.REQUIRED_FIELD : ''}
+                      />
+                    </Grid>
+                  </Grid>
+                ))}
               </>
             )}
-            <Grid
```

Comment:
We need this outside the map({ ... } to show it on 'Out plant' purpose

Reply:
hmmm ok, good catch, thanks. will fix.